### ISSUE TITLE
[Due for payment 2025-03-28] [$250] Add OldDot's default output currency setting to NewDot

### ISSUE THREAD

**Issue by JmillsExpensify** [Expensify/App#52769]
**Problem**
NewDot trial users have consistently run into an issue where they sign up in a certain country – for instance, Spain and EUR – but their employer uses a different currency and expects all reports to be submitted in that same currency. This not only creates confusion, but it's detrimental to our bottom-up conversation overall. 

**Solution**
For consistency with OldDot and to increase our changes that bottom-up flows work without a hitch, let's add a `Payment currency` setting to NewDot. This is equivalent to the `Report Currency` setting in OldDot's `personal` type policy, and functions exactly the same – meaningly, this sets the currency all expenses on a report should be converted to. 

In NewDot the setting will look like this. 
![image](https://github.com/user-attachments/assets/f44d6b60-48ed-4804-b80f-a14e33296d68)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021901571690505630021</li>
        <li>Upwork Job ID: 1901571690505630021</li>
        <li>Last Price Increase: 2025-03-17</li>
<li>Automatic offers: </li>
<ul>
<li>mkzie2 | Contributor | 106569643</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @JmillsExpensify</details>

**Comment by JmillsExpensify** [comment]
@jamesdeanexpensify any feedback on the text on either of these screens? 

Keeping internal until we align on the solution.

**Comment by jamesdeanexpensify** [comment]
Thoughts on:

`Select your currency for paying and getting paid back by friends.`

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Add OldDot's default output currency setting to NewDot

### What is the root cause of that problem?
New feature
### What changes do you think we should make in order to solve the problem?

- The test branch here: https://github.com/mkzie2/App/tree/mkize2-issue/52769-temp

1. Add the payment currency item after the language item [here](https://github.com/Expensify/App/blob/376f9f08755b0555cf3666c5a880542af57c7da1/src/pages/settings/Preferences/PreferencesPage.tsx#L105). To get the current currency we can use `getPersonalPolicy` to get the personal policy and getting the currency from `outputCurrency` field. When clicking on this item we will navigate to the edit payment currency page

```
<MenuItemWithTopDescription
    shouldShowRightIcon
    title={`${paymentCurrency} - ${getCurrencySymbol(paymentCurrency)}`}
    description={translate('billingCurrency.paymentCurrency')}
    onPress={() => Navigation.navigate(ROUTES.SETTINGS_PAYMENT_CURRENCY)}
    wrapperStyle={styles.sectionMenuItemTopDescription}
/>
```
2. Create a `PaymentCurrencyPage` which is the page to edit the currency

- To display the currency list we can use `CurrencySelectionList`
- After selecting a currency, go back to the preference page
- To update the currency we can call `updateGeneralSettings` with `policyID` as the personal policyID


### What alternative solutions did you explore? (Optional)
NA

### Result


https://github.com/user-attachments/assets/369353ac-6bd1-4ee1-abe0-5e1bc18b45bb

**Comment by JmillsExpensify** [comment]
Still in planning

**Comment by JmillsExpensify** [comment]
still in the queue

**Comment by JmillsExpensify** [comment]
Same as above

**Comment by JmillsExpensify** [comment]
Still not a priority

**Comment by JmillsExpensify** [comment]
Still not a priority

**Comment by JmillsExpensify** [comment]
Still holding

**Comment by JmillsExpensify** [comment]
Holding

**Comment by JmillsExpensify** [comment]
@trjExpensify shall I pick this back up? I've been holding on it since it hasn't really come up, but I believe you mentioned it has touched a user fairly recently?

**Comment by trjExpensify** [comment]
Yeah, it was a customer @laurenreidexpensify had. They wanted to change the currency of their distance expense to use a different rate, which is derived from the personal policy currency.

**Comment by JmillsExpensify** [comment]
@trjExpensify is there anything you want to add to the original issue description before I re-open this up?

**Comment by trjExpensify** [comment]
I think it looks good! The only thought that crossed my mind was `Personal currency` vs `Payment currency`, but I see in the description text you clarify it's for personal expenses, so probably fine. 👍

**Comment by JmillsExpensify** [comment]
Ok cool. Going to open this up to the community then!

**Comment by trjExpensify** [comment]
No, the currency on the BNP ("Big number pad") page in the `Create expense` flow shouldn't be changed by this. The `Payment currency` here determines the consolidated currency of the `iouReport` you'll ultimately get paid back in, irrespective of what currency/currencies your expenses were incurred in.

**Comment by trjExpensify** [comment]
Not in this issue, no. 

As for that bug, AFAIK, `Create expense` wherever accessed (FAB or DM menu) should be using the same logic for the currency displayed on the BNP when opening the page. Historically, I think that used a waterfall combo of GPS location, if location services are enabled. If not, the IP address obtained on app load. If not, fallback on the currency of the user’s default workspace. 

It would be good to understand if and where that was changed, as it does look like there's a discrepancy between where `Create expense` is opened for some reason.

**Comment by trjExpensify** [comment]
Oh yeah weird, not sure why someone didn't get assigned. Yup, added you for C+.

**Comment by mountiny** [comment]
@mkzie2 do you need any help from the BE perspective?

**Comment by JmillsExpensify** [comment]
Payment summary: 

 - Contributor: $250 @mkzie2 via Upwork
 - Reviewer: $250 @dukenv0307 via New Expensify

**Comment by JmillsExpensify** [comment]
@dukenv0307 please request via New Expensify. I'm closing this one out!

**Comment by garrettmknight** [comment]
**Payment summary:**

- Contributor: $250 @mkzie2 via Upwork
- Reviewer: $250 @dukenv0307 via New Expensify

**Comment by JmillsExpensify** [comment]
$250 approved for @dukenv0307
